internal/operator/internal/index_image.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"github.com/operator-framework/operator-sdk/internal/operator"
+	registryutil "github.com/operator-framework/operator-sdk/internal/registry"
+)
+
+type IndexImageCatalogCreator struct {
+	IndexImage       string
+	InjectBundles    []string
+	InjectBundleMode string
+
+	cfg *operator.Configuration
+}
+
+func NewIndexImageCatalogCreator(cfg *operator.Configuration) *IndexImageCatalogCreator {
```

Comment:
Any reason `IndexImageCatalogCreator` is exported and this signature isn't
```suggestion
func NewIndexImageCatalogCreator(cfg *operator.Configuration) CatalogCreator {
```
?

Reply:
I'll take a look at this for a follow-up.

Reply:
Actually, one reason is that we could make `NewIndexImageCatalogCreator` work with just an index image. In that case, it could skip the entire registry pod setup and just create the `CatalogSource` with the `spec.index` field set.